packages/pigeon/platform_tests/ios_swift_unit_tests/ios/RunnerTests/EnumTests.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import XCTest
+@testable import Runner
+
+class MockEnumApi2Host: EnumApi2Host {
+  func echo(data: DataWithEnum) -> DataWithEnum {
+    return data
+  }
+}
+
+extension DataWithEnum: Equatable {
+  public static func == (lhs: DataWithEnum, rhs: DataWithEnum) -> Bool {
+    lhs.state == rhs.state
+  }
+}
+
+class EnumTests: XCTestCase {
+  
+  func testEchoHost() throws {
+    let binaryMessenger = MockBinaryMessenger<DataWithEnum>(codec: EnumApi2HostCodec.shared)
+    EnumApi2HostSetup.setUp(binaryMessenger: binaryMessenger, api: MockEnumApi2Host())
+    let channelName = "dev.flutter.pigeon.EnumApi2Host.echo"
+    XCTAssertNotNil(binaryMessenger.handlers[channelName])
+    
+    let input = DataWithEnum(state: .success)
+    let inputEncoded = binaryMessenger.codec.encode([input])
+    
+    let expectation = XCTestExpectation(description: "echo")
+    binaryMessenger.handlers[channelName]?(inputEncoded) { data in
+      let outputMap = binaryMessenger.codec.decode(data) as? [String: Any]
+      XCTAssertNotNil(outputMap)
+      
+      let output = outputMap!["result"] as? DataWithEnum
```

Comment:
avoid force unwrap as much as possible (here and other places)

if this test case fails, we want to continue to the next test case, and not crash the whole test.

Reply:
But in line 34 the test assert the nullability of variable.
Even with this is it better avoid force unwrap?

Reply:
So how `XCTAssert` works is that, if the assert fails, it will record the failure, and continue to the next line. After running all the tests, you get a report of all the failures in all test cases. However, force unwrap a nil value will simply crash the whole test suite. 

For this particular example, when Line 34 fails, it will go to Line 36, and crash the whole test.

Reply:
Fixed